### PR TITLE
Add OpenAI starter sample

### PR DESCRIPTION
**Pull request by Netyyyy**
## Purpose
Add OpenAI starter sample

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: openai/spring-cloud-azure-starter-openai/spring-cloud-azure-openai-sample/src/main/java/com/azure/spring/openai/sample/OpenAIController.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import com.azure.ai.openai.OpenAIClient;
+import com.azure.ai.openai.models.Choice;
+import com.azure.ai.openai.models.Completions;
+import com.azure.ai.openai.models.CompletionsOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
```
Comment (saragluna) [inline openai/spring-cloud-azure-starter-openai/spring-cloud-azure-openai-sample/src/main/java/com/azure/spring/openai/sample/OpenAIController.java:13]: optimize the imports

File: openai/spring-cloud-azure-starter-openai/spring-cloud-azure-openai-sample/src/main/java/com/azure/spring/openai/sample/OpenAIController.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+@RestController
+public class OpenAIController {
+
+    private static final Logger logger = LoggerFactory.getLogger(OpenAIController.class);
+    public static final String deploymentOrModelId = "gpt-35-turbo";
+    @Autowired
+    private OpenAIClient client;
+
+    @PostMapping("/prompt")
+    @ResponseBody
```
Comment (saragluna) [inline openai/spring-cloud-azure-starter-openai/spring-cloud-azure-openai-sample/src/main/java/com/azure/spring/openai/sample/OpenAIController.java:29]: I don't think the `@ResponseBody` is required here https://www.baeldung.com/spring-controller-vs-restcontroller#overview.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: openai/spring-cloud-azure-starter-openai/spring-cloud-azure-openai-sample/src/main/java/com/azure/spring/openai/sample/OpenAIController.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+@RestController
+public class OpenAIController {
+
+    private static final Logger logger = LoggerFactory.getLogger(OpenAIController.class);
+    public static final String deploymentOrModelId = "gpt-35-turbo";
+    @Autowired
+    private OpenAIClient client;
```
Comment (stliu) [inline openai/spring-cloud-azure-starter-openai/spring-cloud-azure-openai-sample/src/main/java/com/azure/spring/openai/sample/OpenAIController.java:25]: we should alway promote spring's best practice in our sample, in this case,, we should use constructor based injection

**Comment by Netyyyy** [comment]
@saragluna , please help take a look, thanks.

**Review by saragluna — APPROVED** [review]

**Comment by backwind1233** [comment]
@Netyyyy FYI this [link](https://github.com/Azure-Samples/azure-spring-boot-samples/pull/39) to you if needed.

**Pull request merged**